clusterloader2/pkg/measurement/common/slos/pod_startup_latency.go

```diff
@@ -129,20 +135,22 @@ func (p *podStartupLatencyMeasurement) start(c clientset.Interface) error {
 	klog.Infof("%s: starting pod startup latency measurement...", p)
 	p.isRunning = true
 	p.stopCh = make(chan struct{})
-	optionsModifier := func(options *metav1.ListOptions) {
-		options.FieldSelector = p.fieldSelector
-		options.LabelSelector = p.labelSelector
-	}
-	listerWatcher := cache.NewFilteredListWatchFromClient(c.CoreV1().RESTClient(), "pods", p.namespace, optionsModifier)
-	p.informer = cache.NewSharedInformer(listerWatcher, &corev1.Pod{}, 0)
-	p.informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc: func(obj interface{}) {
-			p.checkPod(obj)
-		},
-		UpdateFunc: func(oldObj, newObj interface{}) {
-			p.checkPod(newObj)
+	p.informer = informer.NewInformer(
+		c,
+		"pods",
+		p.namespace,
+		p.fieldSelector,
+		p.labelSelector,
+		func(_, newObj interface{}) {
+			if newObj == nil {
+				return
+			}
+			f := func() {
+				p.checkPod(newObj)
+			}
+			p.queue.Add(&f)
```

Comment:
Why do we need queue here?
We didn't have it and I would prefer not to change the logic in refactoring PRs.

Reply:
checkPod method is grabbing lock. I though that maybe when there is a burst of event it can cause watch to lag. Adding a queue would solve the problem - it behaves as a buffer.

Reply:
That's fine - but please avoid changing logic in larger refactoring PR. If you want to do that, please move that change to a separate PR.

Reply:
Ok. I removed the queue.